pkg/virt-launcher/virtwrap/statsconv/converter.go

```diff
@@ -214,9 +214,13 @@ func Convert_libvirt_DomainJobInfo_To_stats_DomainJobInfo(info *libvirt.DomainJo
 	return &stats.DomainJobInfo{
 		DataProcessedSet: info.DataProcessedSet,
 		DataProcessed:    info.DataProcessed,
+		MemoryBpsSet:     info.MemBpsSet,
+		MemoryBps:        info.MemBps,
+		DiskBpsSet:       info.DiskBpsSet,
+		DiskBps:          info.DiskBps,
 		DataRemainingSet: info.DataRemainingSet,
 		DataRemaining:    info.DataRemaining,
-		MemDirtyRateSet:  info.MemDirtyRateSet,
-		MemDirtyRate:     info.MemDirtyRate,
+		MemDirtyRateSet:  info.MemDirtyRateSet && info.MemPageSizeSet,
```

Comment:
if libvirt decides it is not set then it will not be set, and I think in that case we can't assume anything - better not return anything than return the wrong results. I do not think it should happen, but this is the API we have...

Reply:
I still think libvirt should be responsible for this, if there will ever be a problem it should be fixed in the libvirt side, not be masked by what we do here.